halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesClouddriverBootstrapService.java

```diff
@@ -49,7 +50,7 @@ public Settings buildServiceSettings(DeploymentConfiguration deploymentConfigura
     List<String> profiles = new ArrayList<>();
     profiles.add("bootstrap");
     Settings settings = new Settings(profiles);
-    String location = "spinnaker";
+    String location = StringUtils.isEmpty(deploymentConfiguration.getDeploymentEnvironment().getLocation()) ? "spinnaker" : deploymentConfiguration.getDeploymentEnvironment().getLocation();
```

Comment:
If we have it in the deployment environment then we won't be able to get the default per provider since as of now we don't have access to the provider there.